test/ui/BreadcrumbValuesListTest.ts

```diff
@@ -0,0 +1,113 @@
+import { BreadcrumbValueList } from '../../src/ui/Facet/BreadcrumbValuesList';
+import * as Mock from '../MockEnvironment';
+import { Facet } from '../../src/ui/Facet/Facet';
+import { FacetValue } from '../../src/ui/Facet/FacetValues';
+import { BreadcrumbValueElement } from '../../src/ui/Facet/BreadcrumbValueElement';
+import * as _ from 'underscore';
+import { $$ } from '../Test';
+import { FakeResults } from '../Fake';
+import { IIndexFieldValue } from '../../src/rest/FieldValue';
+
+export function BreadcrumbValueListTest() {
+  describe('BreadcrumbValuesListTest', () => {
+    let facet: Facet;
+    let facetValues: FacetValue[];
+
+    beforeEach(() => {
+      facet = Mock.mock(Facet);
+      facet.options = {};
+      facet.options.numberOfValuesInBreadcrumb = 5;
+      facet.getValueCaption = jasmine.createSpy('getValueCaption').and.callFake((facetValue: IIndexFieldValue) => facetValue.value);
+      facetValues = _.map(FakeResults.createFakeFieldValues('foo', 3), fieldValue => {
+        const created = FacetValue.createFromFieldValue(fieldValue);
+        created.selected = true;
+        return created;
+      });
+    });
+
+    it('should support rendering as a simple string and not an HTML element', () => {
+      facet.options.title = 'My facet title';
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
```

Comment:
You could combine `BreadcrumbValueList` initialization and the `build()` call into one line since the variable `breadCrumbList` is not referenced otherwise. This is the case for all the tests in the file.